pkg/logql/parser_test.go

```diff
@@ -534,19 +534,143 @@ func TestParse(t *testing.T) {
 		{
 			in: `{foo="bar"} "foo"`,
 			err: ParseError{
-				msg:  "syntax error: unexpected STRING, expecting != or !~ or |~ or |=",
+				msg:  "syntax error: unexpected STRING",
 				line: 1,
 				col:  13,
 			},
 		},
 		{
 			in: `{foo="bar"} foo`,
 			err: ParseError{
-				msg:  "syntax error: unexpected IDENTIFIER, expecting != or !~ or |~ or |=",
+				msg:  "syntax error: unexpected IDENTIFIER",
 				line: 1,
 				col:  13,
 			},
 		},
+		{
+			// require left associativity
+			in: `
+sum(count_over_time({foo="bar"}[5m])) by (foo) /
```

Comment:
I'd like to have more failure tests, specially around unexpected type like two logstreams (`{app="foo"} + {app="fuzz"}`) which you seem to support reading the yacc file. (I might be wrong).

This would also be nice for  covering panic thrown in the ast.go

We should try to keep error message for those bad query useful for our users.